File: solutions/ruby/resistor-color/1/resistor_color.rb
module ResistorColor

  COLORS = %w[
    black
    brown
    red
    orange
    yellow
    green
    blue
    violet
    grey
    white
  ]

  def self.color_code(color)
    COLORS.index(color)
  end

  def self.colors
    COLORS
  end

end

File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t band) {
    return (uint16_t)band;
}

const resistor_band_t* colors(void) {
    static const resistor_band_t colors[] = {
        BLACK, BROWN, RED, ORANGE, YELLOW,
        GREEN, BLUE, VIOLET, GREY, WHITE
    };
    return colors;
}
